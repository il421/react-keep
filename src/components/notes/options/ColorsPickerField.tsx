import React from "react";
import { FlexBox } from "../../ui-components/FlexBox";
import { JustifyContent, PickerColors } from "../../../common/variables";
import "../../../styles/components/notes/_colors-picker.scss";
import { RadioButtonsInputField } from "../../form/RadioButtonsInputField";

class ColorsPickerField extends React.PureComponent {
  render() {
    return (
      <FlexBox
        justifyContent={JustifyContent.spaceBetween}
        className="colors-picker"
      >
        {
          Object.values(PickerColors).map((color: string) => (
            <RadioButtonsInputField
              key={ color }
              name="color"
              radioClassName="colors-picker__item"
              value={ color }
              labelProps={{
                style: { backgroundColor: color, cursor: "pointer" },
              }}
            />
          ))
        }
      </FlexBox>
    );
  }
}

export default ColorsPickerField;
