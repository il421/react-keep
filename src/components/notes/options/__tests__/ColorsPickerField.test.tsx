import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import ColorsPickerField from "../ColorsPickerField";

test("should render ColorsPickerField correctly", () => {
  const wrapper: ShallowWrapper = shallow(<ColorsPickerField />);
  expect(wrapper.debug()).toMatchSnapshot();
});
