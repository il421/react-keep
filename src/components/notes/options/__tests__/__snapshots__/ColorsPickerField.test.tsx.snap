// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ColorsPickerField correctly 1`] = `
"<FlexBox justifyContent=\\"space-between\\" className=\\"colors-picker\\">
  <RadioButtonsInputField id=\\"color-icon-#fff\\" name=\\"color\\" radioClassName=\\"colors-picker__item\\" value=\\"#fff\\" labelProps={{...}} />
  <RadioButtonsInputField id=\\"color-icon-#f28b82\\" name=\\"color\\" radioClassName=\\"colors-picker__item\\" value=\\"#f28b82\\" labelProps={{...}} />
  <RadioButtonsInputField id=\\"color-icon-#cbf0f8\\" name=\\"color\\" radioClassName=\\"colors-picker__item\\" value=\\"#cbf0f8\\" labelProps={{...}} />
  <RadioButtonsInputField id=\\"color-icon-#fff475\\" name=\\"color\\" radioClassName=\\"colors-picker__item\\" value=\\"#fff475\\" labelProps={{...}} />
  <RadioButtonsInputField id=\\"color-icon-#ccff90\\" name=\\"color\\" radioClassName=\\"colors-picker__item\\" value=\\"#ccff90\\" labelProps={{...}} />
  <RadioButtonsInputField id=\\"color-icon-#e6c9a8\\" name=\\"color\\" radioClassName=\\"colors-picker__item\\" value=\\"#e6c9a8\\" labelProps={{...}} />
  <RadioButtonsInputField id=\\"color-icon-#e8eaed\\" name=\\"color\\" radioClassName=\\"colors-picker__item\\" value=\\"#e8eaed\\" labelProps={{...}} />
  <RadioButtonsInputField id=\\"color-icon-#fbbc04\\" name=\\"color\\" radioClassName=\\"colors-picker__item\\" value=\\"#fbbc04\\" labelProps={{...}} />
</FlexBox>"
`;
